src/Interpreters/executeQuery.cpp

```diff
@@ -712,7 +719,8 @@ static std::tuple<ASTPtr, BlockIO> executeQueryImpl(
                     elem.query_views = info.views;
                 }
 
-                interpreter->extendQueryLogElem(elem, ast, context, query_database, query_table);
+                if (interpreter)
+                    interpreter->extendQueryLogElem(elem, ast, context, query_database, query_table);
```

Comment:
btw why does it work if now as I see `interpreter` is `nullptr` in case `async_insert == true`?

Reply:
What do you mean? `extendQueryLogElem` just adds table and databases used in query. I think, we can add element to query log without this info. However, let's try to keep it in case of async inserts..